src/ResourceManager/DataFactories/Commands.DataFactoryV2/DataFactories/UpdateAzureDataFactoryCommand.cs

```diff
@@ -0,0 +1,94 @@
+using System.Collections;
+using System.Globalization;
+using System.Management.Automation;
+using System.Security.Permissions;
+using Microsoft.Azure.Commands.DataFactoryV2.Models;
+using Microsoft.Azure.Commands.DataFactoryV2.Properties;
+using Microsoft.Azure.Management.DataFactory.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.DataFactoryV2
+{
+    [Cmdlet(VerbsData.Update, Constants.DataFactory, DefaultParameterSetName = ParameterSetNames.ByFactoryName, SupportsShouldProcess = true),
+        OutputType(typeof(PSDataFactory))]
+    public class UpdateAzureDataFactoryCommand : DataFactoryBaseCmdlet
+    {
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryName, Position = 0, Mandatory = true,
+            HelpMessage = Constants.HelpResourceGroup)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryName, Position = 1, Mandatory = true,
+            HelpMessage = Constants.HelpFactoryName)]
+        [Alias(Constants.DataFactoryName)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryObject, Position = 0, Mandatory = true, ValueFromPipeline = true,
+            HelpMessage = Constants.HelpFactoryObject)]
+        [ValidateNotNullOrEmpty]
+        public PSDataFactory InputObject { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ByResourceId, Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.HelpResourceId)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryName, Position = 2, Mandatory = false,
+            HelpMessage = Constants.HelpTagsForFactory)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryObject, Position = 1, Mandatory = false,
+            HelpMessage = Constants.HelpTagsForFactory)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByResourceId, Position = 1, Mandatory = false,
+            HelpMessage = Constants.HelpTagsForFactory)]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryName, Position = 2, Mandatory = false, ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.HelpIdentityForFactory)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryObject, Position = 1, Mandatory = false, ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.HelpIdentityForFactory)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByResourceId, Position = 1, Mandatory = false, ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.HelpIdentityForFactory)]
+        public FactoryIdentity Identity { get; set; }
```

Comment:
@hvermis where is a user supposed to get this value from?

Reply:
This is one of the properties on data factory which gets created when they create a factory. The class is part of our SDK, and they can create a new instance of that class and pass it to update command.

Reply:
I changed the Value from pipeline to be by object. Here's an exampl of how it would look like:
(Get-AzureRmDataFactoryv2 -ResourceGroupName hvtest -DataFactoryName hvadf1).Identity|Update-AzureRmDataFactoryV
2 -DataFactoryName hvadf -ResourceGroupName hvtest

Is this worth supporting? Should I just remove the ValueFromPipeline completely?

Reply:
@hvermis there currently exists a parameter set where `DataFactory` and `Identity` are in the same parameter set and both accept their value from the pipeline, which shouldn't happen. Only `DataFactory` should accept its value from the pipeline the in the parameter set, and we should look at ways of allowing the user to update the value of `Identity`.